### DISCUSSION TITLE
Create multiple releases of mariadb-alpine?

### DISCUSSION THREAD

**Discussion by Luc45** [jbergstroem/mariadb-alpine#230] [Ideas]
Howdy, is `mysqldump` available in this image? Would it add too much size to add it?

**Comment by Luc45** [comment]
Opened a PR to see how many kbs it adds to the image: https://github.com/jbergstroem/mariadb-alpine/pull/229

**Comment by jbergstroem** [comment]
Thanks for opening the PR! As you also see; finding the middle ground between functionality and minimalism is hard.

Seeing how this is more of a client tool rather than server; I'm really on the fence.

**Comment by jbergstroem** [comment]
Old topic: `mysqldump` is not included?

I took the liberty of moving this issue to a discussion. The topic is "which tools would a extended version of the container include".

**Comment by jbergstroem** [comment]
How I see this:

- I would like to keep the baseline / default release as small as possible which would be as-is (including a client). The core purpose is to distribute a working server
- The extended tag would contain additional binaries. What I'd consier at this point: `mysqldump`, `mysqladmin`, `aria_chk`, `myisamchk`, `mysqlcheck`, `mysql_upgrade` (from `mariadb-server-utils`) and `mysql_tzinfo_to_sql`.

Thoughts?

Edit: added `mysql_tzinfo_to_sql` to simplify the [tz population](https://github.com/jbergstroem/mariadb-alpine/blob/main/docs/usage.md#import-tzdata).

**Reply by williamdes** [reply]
That would be awesome !

But why not do the opposite ?

Do a full version and do a `-slim` tag like most Docker images do ?

**Reply by jbergstroem** [reply]
The goal of this project is to optimize for size, not for feature parity. There are lots of containers that gives you a fully fledged mariadb setup with all tools you need; I don't see the reason to create another one.

**Reply by Luc45** [reply]
I agree with @jbergstroem here. The main point of this image is the size. About which programs to include, it would be nice to know the size of each in kb to decide.

**Reply by jbergstroem** [reply]
> , it would be nice to know the size of each in kb to decide.

Would you say that size is still important in the `-extended` or `-tools` or w/e we end up calling it?

**Reply by williamdes** [reply]
`-tools` would make sense

**Reply by williamdes** [reply]
the list looks correct
